site/source/tutorials/003_creating_widgets/index.md

```diff
@@ -237,25 +243,94 @@ export default class WorkerContainer extends WorkerContainerBase<ThemeableProper
+
+Let's start by moving the `Worker` data into the `App` class. First, add the following import to the `App.ts` file:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' line:6 %}
+
+Notice that we are going to be using the `WorkerProperties` interface, not the `Worker` widget. We want the `App` class to work with the **data** that describes the application's state, not the **widgets** that will render it. This keeps our application more flexible my deferring the decision about how to render the `WorkerProperties` data to a lower-level widget in our application.
+
+Now it's time to specify the worker data. Add this code to the top of the `App` class definition:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:9-22 %}
+
+This data is the same as what we had previous with exception of the `key` property. The keys should be added by whatever widget is rendering the `Worker`s so that it can ensure that each one receives a unique value.
+
+The last thing we need to do in this file is pass the worker data into the `WorkerContainer`. Update the `render` method to do that:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:24-31 %}
+
+Now, let's update the `WorkerContainer.ts` file to be able to accept and render this data.
+
+Start by updating the import statement that is retrieving the `Worker` class to import the `WorkerProperties` interface as well:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' line:4 %}
+
+Now we need to create a new interface that extends `ThemeableProperties` to allow the `WorkerContainer` to accept an array of `WorkerProperties`. Add these lines:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' lines:8-10 %}
+
+Next, update the class declaration to work with the new `WorkerContainerProperties` interface:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' line:15 %}
+
+Finally, update the `render` method to remove the previous, hard-coded widgets and replace that will code that will dynamically generate `Worker`s based on the `workerData` property:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' lines:16-29 %}
+
+Refactoring the application has made the `WorkerContainer` more flexible than the initial implementation. This is because it is no longer responsible for determining where its data comes from. It can receive data from any external provider, such as the `App` class or a data store, and renders it.
```

Comment:
"because it is no longer responsible for determining where its data comes from" -> "because it is no longer responsible for determining its data source"